### DISCUSSION TITLE
rounded corners without CSD

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [YaLTeR/niri#164] [Ideas]
the [code now notes](https://github.com/YaLTeR/niri/blob/fbe902091533c211a8e61905951fdd113fbca11e/src/handlers/xdg_shell.rs#L43C35-L44):

> If the user prefers no CSD, it's a reasonable assumption that they would prefer to get rid of the various client-side rounded corners also by using the tiled state.

i have been interested in using rounded corners without using top bars, see e.g. [here](https://imgur.com/LPC6uBf).
it would be nice if rounding were available as a setting independent of other such options.

**Comment by YaLTeR** [comment]
Yeah, this will need compositor-side rounding. I'm planning to give it a try at some point.

**Comment by cptn-cosmo** [comment]
would also love to have rounded corners :)

**Comment by Connor-GH** [comment]
Same here. Features that prevent me from switching from Hyprland include: lack of mouse hover-over focus, no rounded corners, and not many animations.

**Reply by YaLTeR** [reply]
Animations are coming, they just take a while and some depend on a lot of refactoring to get right. :smile: I pushed window opening animations recently.

**Comment by YaLTeR** [comment]
This has been implemented and released in v0.1.6: https://github.com/YaLTeR/niri/releases/tag/v0.1.6
